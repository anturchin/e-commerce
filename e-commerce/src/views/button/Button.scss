.button {
    margin-top: 5px;
    padding: 10px 25px;
    border: none;
    border-radius: 5px;
    color: #ffffff;
    font-size: 1.6rem;
    font-weight: 600;
    background-color: #0f0;
    transition: all 250ms;
    outline: none;
    cursor: pointer;
    &:hover:not(:disabled),
    &:focus:not(:disabled) {
        background-color: #0f0;
    }
    &:hover:not(:disabled) {
        transform: translateY(-1px);
        background-color: rgb(4, 164, 4);
    }
    &:disabled {
        opacity: 0.3;
        cursor: auto;
    }
}
