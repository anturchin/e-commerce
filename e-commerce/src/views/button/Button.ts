import { View } from '../View';
import { ButtonType } from './types';

import './Button.scss';

export class Button extends View {
    constructor(props: ButtonType) {
        super({ tag: 'button', classNames: ['custom-button'] });
        this.setupInput(props);
        this.addClickHandler(props.onClick);
    }

    private setupInput(props: ButtonType): void {
        const button = this.getElement() as HTMLButtonElement;
        button.textContent = props.label;
        if (props.disabled) {
            button.disabled = props.disabled;
        }
    }

    public addClickHandler(onClick?: () => void): void {
        if (onClick) {
            const button = this.getElement() as HTMLButtonElement;
            button.addEventListener('click', onClick);
        }
    }
}
