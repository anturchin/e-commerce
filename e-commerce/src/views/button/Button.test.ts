import { describe, expect, test, jest } from '@jest/globals';
import { Button } from './Button';

describe('Button', () => {
    test('should match', () => {
        const onClickMock = jest.fn();
        const props = {
            label: 'Click Me',
            onClick: onClickMock,
            disabled: false,
        };

        const button = new Button(props);
        const renderedButton = button.getElement();
        expect(renderedButton.outerHTML).toMatchSnapshot();
    });
});
