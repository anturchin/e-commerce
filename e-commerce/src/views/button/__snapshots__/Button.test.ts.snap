// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match 1`] = `"<button class="button">Click Me</button>"`;
